src/coreclr/src/vm/diagnosticsserverprotocolhelper.h

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __DIAGNOSTICSSERVER_PROTOCOL_HELPER_H__
+#define __DIAGNOSTICSSERVER_PROTOCOL_HELPER_H__
+
+#ifdef FEATURE_PERFTRACING
+
+#include "common.h"
+#include "diagnosticsipc.h"
+#include "diagnosticsprotocol.h"
+
+class IpcStream;
+
+/**
+ * The Diagnostics Server command set is 0xFF
+ * see diagnosticsipc.h and diagnosticserver.h for more details
+ * enum class DiagnosticServerCommandId : uint8_t
+ * {
+ *     OK    = 0x00,
+ *     Error = 0xFF,
+ *     Advertise = 0x01,
```

Comment:
Is there an updated protocol document somewhere that describes the protocol in enough detail that someone could implement it? Last I looked the design doc didn't have that level of detail.

Reply:
The documentation for this lives in dotnet/diagnostics.  I will update that file accordingly.  For now, I'm treating the Advertise message as a separate, meta protocol.  It will be documented.